ksqldb-engine/src/main/java/io/confluent/ksql/topic/TopicFactory.java

```diff
@@ -51,4 +56,14 @@ public static KsqlTopic create(final CreateSourceProperties properties) {
         valueFormat
     );
   }
+
+  private static FormatInfo getDefaultKeyFormat(final KsqlConfig config) {
+    final String defaultKeyFormat = config.getString(KsqlConfig.KSQL_DEFAULT_KEY_FORMAT_CONFIG);
+    if (defaultKeyFormat == null) {
+      throw new KsqlException("Blah blha");
+    }
+
+    // Would need to wire in key properties obvs
+    return FormatInfo.of(defaultKeyFormat);
```

Comment:
This is the code that handles the default key format config for C* statements - you'd need to extend for default value format.